src/test/java/org/broadinstitute/hellbender/utils/pileup/ReadPileupUnitTest.java

```diff
@@ -0,0 +1,345 @@
+        Assert.assertEquals(rg2Reads.get(2), read5, "Read " + read5.getName() + " should be in rg2 but isn't");
+    }
+
+    @Test(expectedExceptions = GATKException.class)
+    public void testException1() {
+        final SAMFileHeader header = ArtificialReadUtils.createArtificialSamHeader(1,1,1000);
+
+        final GATKRead read1 = ArtificialReadUtils.createArtificialRead(header,"read1",0,1,10);
+
+        final ReadPileup pileup = new ReadPileup(null,
+                null,
+                Arrays.asList(1));
+    }
+
+    @Test(expectedExceptions = GATKException.class)
+    public void testException2() {
+        final SAMFileHeader header = ArtificialReadUtils.createArtificialSamHeader(1, 1, 1000);
+
+        final GATKRead read1 = ArtificialReadUtils.createArtificialRead(header,"read1",0,1,10);
+
+        final ReadPileup pileup = new ReadPileup(null,
+                Arrays.asList(read1),
+                null);
+    }
+
+    @Test(expectedExceptions = GATKException.class)
+    public void testException3() {
+        final SAMFileHeader header = ArtificialReadUtils.createArtificialSamHeader(1, 1, 1000);
+
+        final GATKRead read1 = ArtificialReadUtils.createArtificialRead(header,"read1",0,1,10);
+
+        final ReadPileup pileup = new ReadPileup(null,
+                Arrays.asList(read1),
+                Arrays.asList(1, 2, 3));
+    }
+
+
+    /**
+     * Ensure that splitting read groups still works when dealing with null read groups.
+     */
+    @Test
+    public void testSplitByNullReadGroups() {
+        final SAMFileHeader header = ArtificialReadUtils.createArtificialSamHeader(1,1,1000);
+
+        final GATKRead read1 = ArtificialReadUtils.createArtificialRead(header,"read1",0,1,10);
+        final GATKRead read2 = ArtificialReadUtils.createArtificialRead(header,"read2",0,1,10);
+        final GATKRead read3 = ArtificialReadUtils.createArtificialRead(header,"read3",0,1,10);
+
+        final ReadPileup pileup = new ReadPileup(null,
+                                                           Arrays.asList(read1, read2, read3),
+                                                           Arrays.asList(1, 1, 1));
+
+        final ReadPileup nullRgPileup = pileup.getPileupForReadGroup(null);
+        final List<GATKRead> nullRgReads = nullRgPileup.getReads();
+        Assert.assertEquals(nullRgPileup.size(), 3, "Wrong number of reads in null read group");
+        Assert.assertEquals(nullRgReads.get(0), read1, "Read " + read1.getName() + " should be in null rg but isn't");
+        Assert.assertEquals(nullRgReads.get(1), read2, "Read " + read2.getName() + " should be in null rg but isn't");
+        Assert.assertEquals(nullRgReads.get(2), read3, "Read " + read3.getName() + " should be in null rg but isn't");
+
+        final ReadPileup rg1Pileup = pileup.getPileupForReadGroup("rg1");
+        Assert.assertNull(rg1Pileup, "Pileup for non-existent read group should return null");
+    }
+
+    private static int sampleI = 0;
```

Comment:
Avoid static variables in tests -- put these into data providers or the test cases themselves.